DCF_model.py

```diff
@@ -0,0 +1,117 @@
+"""
+@project       : Queens College CSCI 365/765 Computational Finance
+@Instructor    : Dr. Alex Pang
+
+@Student Name  : first last
+
+@Date          : June 2021
+
+Discounted Cash Flow Model with Financial Data from Yahoo Financial
+
+https://github.com/JECSand/yahoofinancials
+
+
+"""
+import enum
+import calendar
+import math
+import pandas as pd
+import numpy as np
+
+import datetime 
+from scipy.stats import norm
+
+from math import log, exp, sqrt
+
+from stock import Stock
+
+
+class DiscountedCashFlowModel(object):
+    """
+    DCF Model:
+
+    FCC is assumed to go have growth rate by 3 periods, each of which has different growth rate
+           short_term_growth_rate for the next 5Y
+           medium_term_growth_rate from 6Y to 10Y
+           long_term_growth_rate from 11Y to 20thY
+    """
+
+    def __init__(self, stock, as_of_date):
+        self.stock = stock
+        self.as_of_date = as_of_date
+
+        self.short_term_growth_rate = None
+        self.medium_term_growth_rate = None
+        self.long_term_growth_rate = None
+
+    def set_FCC_growth_rate(self, short_term_rate, medium_term_rate, long_term_rate):
+        self.short_term_growth_rate = short_term_rate
+        self.medium_term_growth_rate = medium_term_rate
+        self.long_term_growth_rate = long_term_rate
+
+    def calc_fair_value(self):
+        """
+        calculate the fair_value using DCF model as follows
+
+        1. calculate a yearly discount factor using the WACC
+        2. Get the Free Cash flow
+        3. Sum the discounted value of the FCC for the 20 years using similar approach as presented in class
+        4. Compute the PV as cash + short term investments - total debt + the above sum of discounted free cash flow
+        5. Return the stock fair value of the stock
+        """
+        
+        #TODO
+
+        WACC = self.stock.lookup_wacc_by_beta(self.stock.get_beta())
+        DF = 1 / (1 + WACC)
+
+        FCC = self.stock.get_free_cashflow()
+        STR = self.short_term_growth_rate
+        MTR = self.medium_term_growth_rate
+        LTR = self.long_term_growth_rate
+
+        DCF = 0
+        for i in range(1, 6):
+            DCF += FCC * ((1 + STR) ** i) * (DF ** i)
+        CF5 = FCC * ((1 + STR) ** 5)
+
+        for i in range(1, 6):
+            DCF += CF5 * ((1 + MTR) ** i) * (DF ** (i + 5))
+        CF10 = CF5 * ((1 + MTR) ** 5)
+
+        for i in range(1, 11):
+            DCF += CF10 * ((1 + LTR) ** i) * (DF ** (i + 10))
+            
+        PV = self.stock.get_cash_and_cash_equivalent() - self.stock.get_total_debt() + DCF

```

Comment:
Isn't it cash + `short term investments` - total debt + DCF?

Reply:
not short-term investments because it's 20 years?
Code from Professor's example: PV = current_cash - total_debt + DCF

Reply:
![image](https://user-images.githubusercontent.com/55901916/141354430-1d9d9bfb-398e-45f9-afa2-21cfff1ad149.png)
This is what I found on slide, and also see step 4 above (`4. Compute the PV as cash + short term investments - total debt + the above sum of discounted free cash flow`)